### PR TITLE
Add maintainer and contributor documentation about workflow automation near the configuration

### LINKED ISSUE #7364: Add documentation in the napari repo about the GitHub actions workflows and templates
## 📚 Documentation

When debugging action workflow results, having some contributor/maintainer documentation in the `.github` directory is helpful.

Document the actionlint checks, constraints workflows, and info about automation to reduce manual toil.

### PR DESCRIPTION
**Pull request by willingc**
Closes #7364 

# Description

This PR adds a CONTRIBUTING.md to the `.github` directory. The purpose is to quickly
orient a maintainer or contributor about our testing actions, constraints workflows, and
general automation of labels and issues.

### CODE REVIEW

**Comment by melissawm** [comment]
Interesting! I had never thought of doing this but it does make sense. Note we have some of the docs about the docs deployment here: https://napari.org/stable/developers/contributing/documentation/docs_deployment.html

I wonder if we should point people to that page, or move some of that content here? My fear is duplication and going out of sync...

**Comment by willingc** [comment]
@melissawm On projects with a bunch of workflows, it can be helpful to have a README or CONTRIBUTING in the repo to orient quickly people investigating the automation. I think adding a link back to the more general contributor documentation makes sense. I see this files as being navigation or quick orientation for experienced contributors to get context quickly. I view this doc as a "what" and the napari documentation as a "how".

**Review by psobolewskiPhD — APPROVED** [review]
I like this. I think having this near the source makes a lot of sense.
Thank you @willingc

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+## Workflows and actions
+
+There are over 20 GitHub workflows found in `.github/workflows`.
+The team creates a workflow to automate manual actions and steps.
+This results in improved accuracy and quality. Some key workflows:
+- `actionlint.yml` does static testing of GitHub action workflows
+- benchmarks
+- `reusable_run_tox_test.yml` uses our constraint files to install the
+  compatible dependencies for each test environment which may differ
+  by OS and qt versions.
```
Comment (Czaki) [inline .github/CONTRIBUTING.md]:
```suggestion
- `reusable_run_tox_test.yml` uses our constraint files to install the
  compatible dependencies for each test environment which may differ
  by OS and qt versions. It is called from `test_pull_request.yml` and `test_comprehensive.yml`, not directly. 
```

**Comment by willingc** [comment]
Let's press the green button and get this merged. Thanks!

**Comment by melissawm** [comment]
Done 🚀

**Pull request merged**